14 спринт

David-Arcanone opened the pull request:
- дебаунс на нажатие трека, проверял поставив коммент на интент открытия плеера
- многопоточность в экране поиска у запроса с помощью запуска нового Thread
- у некоторых файлов нет пробника на 30 сек в iTunes (может это фильмы/подкасты), в таком случае вместо таймера у меня текст с предупреждением

David-Arcanone committed:
задание спринта 14

David-Arcanone committed:
задание спринта 14

David-Arcanone committed:
задание спринта 14
убрал опечатки

i-masloed commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -64,76 +78,123 @@ class SearchActivity : AppCompatActivity() {
-                                //записаны данные и они не null
-                                myTracks.addAll(response.body()?.results!!)
-                                trackAdapter.notifyDataSetChanged()
+            synchronized(myTracks) { myTracks.clear() }
```
если изменять myTracks только из главного потока, то synchronized избыточен

i-masloed commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -64,76 +78,123 @@ class SearchActivity : AppCompatActivity() {
+
+                        override fun onFailure(call: Call<iTunesSearchResponse>, t: Throwable) {
+                            //произошла ошибка
+                            synchronized(myTracks) {
```
synchronized здесь избыточен

David-Arcanone committed:
задание спринта 14
добавил установку таймера на 00:00 после проигрывания
снизил количество вызовов synchronized

David-Arcanone committed:
задание спринта 14
убрал избыточный synchronized

PS: почему в пункте дебаунс нажатия элемента списка "✅❌", нет пояснения. Дебаунс реализован в "mytrackClicker" который я привязываю к адаптеру истории треков и к адаптеру результатов поиска.
